x-pack/legacy/plugins/infra/public/components/inventory/toolbars/host_toolbar.tsx

```diff
@@ -0,0 +1,129 @@
+      defaultMessage: 'Memory usage',
+    });
+
+    const InboundTraffic = i18n.translate('xpack.infra.waffle.metricOptions.inboundTrafficText', {
+      defaultMessage: 'Inbound traffic',
+    });
+
+    const OutboundTraffic = i18n.translate('xpack.infra.waffle.metricOptions.outboundTrafficText', {
+      defaultMessage: 'Outbound traffic',
+    });
+
+    const LogRate = i18n.translate('xpack.infra.waffle.metricOptions.hostLogRateText', {
+      defaultMessage: 'Log rate',
+    });
+
+    const Load = i18n.translate('xpack.infra.waffle.metricOptions.loadText', {
+      defaultMessage: 'Load',
+    });
+
+    return [
+      {
+        text: CPUUsage,
+        value: InfraSnapshotMetricType.cpu,
+      },
+      {
+        text: MemoryUsage,
+        value: InfraSnapshotMetricType.memory,
+      },
+      {
+        text: Load,
+        value: InfraSnapshotMetricType.load,
+      },
+      {
+        text: InboundTraffic,
+        value: InfraSnapshotMetricType.rx,
+      },
+      {
+        text: OutboundTraffic,
+        value: InfraSnapshotMetricType.tx,
+      },
+      {
+        text: LogRate,
+        value: InfraSnapshotMetricType.logRate,
+      },
+    ];
+  }, []);
+
+  const groupByOptions = useMemo(() => {
+    const mapFieldToOption = (field: string) => ({
+      text: fieldToName(field),
+      field,
+    });
+    return [
+      'cloud.availability_zone',
+      'cloud.machine.type',
+      'cloud.project.id',
+      'cloud.provider',
+      'service.type',
+    ].map(mapFieldToOption);
+  }, []);
+
+  return (
+    <Toolbar>
+      <EuiFlexGroup alignItems="center" gutterSize="m">
+        <WithSource>
+          {({ createDerivedIndexPattern }) => (
+            <WithWaffleOptions>
+              {({
+                changeMetric,
+                changeGroupBy,
+                changeCustomOptions,
+                customOptions,
+                groupBy,
+                metric,
+                nodeType,
+              }) => (
+                <>
+                  <EuiFlexItem grow={false}>
+                    <WaffleMetricControls
+                      options={metricOptions}
+                      metric={metric}
+                      onChange={changeMetric}
+                    />
+                  </EuiFlexItem>
+                  <EuiFlexItem grow={false}>
+                    <WaffleGroupByControls
+                      options={groupByOptions}
+                      groupBy={groupBy}
+                      nodeType={nodeType}
+                      onChange={changeGroupBy}
+                      fields={createDerivedIndexPattern('metrics').fields}
+                      onChangeCustomOptions={changeCustomOptions}
+                      customOptions={customOptions}
+                    />
+                  </EuiFlexItem>
+                </>
+              )}
+            </WithWaffleOptions>
+          )}
+        </WithSource>
+      </EuiFlexGroup>
+    </Toolbar>
```

Comment:
Yeah, I agree about the boilerplate. I didn't go with 1 toolbar with props, because I'm afraid that that component might become complicated and less flexible. As of now, moving to 1 toolbar would only add a couple of if statements, but with the addition of the AWS toolbar, we will need a conditional dropdown for `region`. What if I just created a `ToolbarWrapper` component that contains all of the boilerplate and left each of the Toolbar components to specify the toolbar items?

The toolbar wrapper could look something like this:

``` ts  
<Toolbar>
      <EuiFlexGroup alignItems="center" gutterSize="m">
        <WithSource>
          {({ createDerivedIndexPattern }) => (
            <WithWaffleOptions>
              {({
                changeMetric,
                changeGroupBy,
                changeCustomOptions,
                customOptions,
                groupBy,
                metric,
                nodeType,
              }) => (
```

Reply:
Also, could centralize all of the translations since into the ToolbarWrapper since some of those are duplicated too.